src/AppInstallerCLICore/Workflows/DownloadFlow.cpp

```diff
@@ -458,11 +458,11 @@ namespace AppInstaller::CLI::Workflow
     {
         if (context.Contains(Execution::Data::InstallerPath))
         {
-            if (WI_IsFlagSet(context.GetFlags(), Execution::ContextFlag::InstallerTrusted))
+            if (WI_AreAllFlagsSet(context.GetFlags(), Execution::ContextFlag::InstallerTrusted | Execution::ContextFlag::InstallerHashMatched))

```

Comment:
Actually InstallerTrusted is only set when hash matches.

Reply:
Oh, right. I got that wrong in the description. But still I think it would be good to check here as a precaution in case that code ever changes.

Reply:
And for hash mismatch case, we leave the Motw as URLZONE_INTERNET. When we create the temp file before downloading, we set the Motw.